### PR TITLE
MKT-2022/doc-v1.6.0

### PR DESCRIPTION
**Pull request by Kaushty**
- Add docs for changes made in v1.6.0
- Add docs for older missing methods.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,12 +26,47 @@ Below we have listed some of the top-level objects used in the App SDK.
...
+-   **[Modal](#Modal)**: TIt's a class that represents a modal dialog opened from the app within the Contentstack UI.
+
+# Top level Methods
+
+The App SDK provides several top-level methods that retrieve app-related information.
+
+-   **getConfig**: Retrieves the configuration set for the app. This method allows easy access to the app's configuration data set on the app configuration page.
+-   **getCurrentLocation**: Returns the current UI location type of the app.
+-   **getCurrentRegion**: : Retrieves the Contentstack Region on which the app is running.
+-   **getAppVersion**: Returns the version of the app currently running.
```
Comment (Kaushty) [inline README.md:41]: currently installed.

File: README.md [hunk 2]
```diff
@@ -41,7 +76,7 @@ Locations refers to the position or the placement of the app (sidebar widget, cu
 -   **[DashboardWidget](#DashboardWidget)**: It's an object representing the Dashboard widget reference in the Contentstack UI.
 -   **[SidebarWidget](#SidebarWidget)**: It's an object representing the current Sidebar widget reference in the Contentstack UI.
 -   **[AppConfigWidget](#AppConfigWidget)**: It's an object representing the current App configuration for the current App in the Contentstack UI.
--   **[EntryfieldLocation](#EntryFieldLocation)**: It's an object representing the current Entry field reference over the field in the Contentstack UI.
+-   **[FieldModifierLocation](#FieldModifierLocation)**: It's an object representing the field modifier reference over the field in the Contentstack UI.
```
Comment (Kaushty) [inline README.md:79]: Field Modifier

File: README.md [hunk 3]
```diff
@@ -623,6 +680,27 @@ This method disables the auto resizing of the extension height.
 
 **Returns**: [frame](#frame).
 
+#### frame.preventFrameClose(state: boolean) ⇒ [Promise\<void>](#external_Promise)
+
+The `frame.preventFrameClose` method allows you to manage the default closing behavior of the app when the user clicks outside of its frame. By default, the app is closed in such scenarios. This method enables you to control and customize the closing behavior based on your requirements.
```
Comment (Kaushty) [inline README.md:685]: "clicks outside its frame"

File: README.md [hunk 4]
```diff
@@ -2942,3 +3020,23 @@ IMetadateDelete {
...
+
+## Modal
+
+The `Modal` class represents a modal dialog opened from the app within the Contentstack UI. It is a powerful feature of the App SDK that enables apps to open modal dialogues, providing an enhanced user experience.
+
+> **Note**: Starting from v1.6.0 of the App SDK, modals now open to take the full screen by default, without any additional user action.
+
+### `setBackgroundElement(element: HTMLElement)`
+
+This method allows developers to specify a custom HTML element to be displayed in the background in place of the app iframe when the modal is opened. By default, the App SDK automatically selects an element to be shown in the background. However, this method provides the flexibility to choose a different element if desired.
```
Comment (Kaushty) [inline README.md:3032]: "element to be displayed in the background, in place of .."
"different element if required."

File: README.md [hunk 5]
```diff
@@ -2942,3 +3020,23 @@ IMetadateDelete {
 ```
 
 This method deletes existing metadata for an asset or entry. It accepts metadata configuration as required arguments. This config contains basic details that you need to identify the metadata object you want to delete.
+
+## Modal
+
+The `Modal` class represents a modal dialog opened from the app within the Contentstack UI. It is a powerful feature of the App SDK that enables apps to open modal dialogues, providing an enhanced user experience.
```
Comment (Kaushty) [inline README.md:3026]: "This feature of the App SDK enables apps to open modal dialogues, providing an enhanced user experience." in place of "It is a powerful feature of the App SDK that enables apps to open modal dialogues, providing an enhanced user experience."

File: README.md (hunk 1 above)
Comment (Kaushty) [reply README.md]: Resolved

File: README.md (hunk 2 above)
Comment (Kaushty) [reply README.md]: Addressed.

File: README.md (hunk 3 above)
Comment (Kaushty) [reply README.md]: Addressed.

File: README.md (hunk 4 above)
Comment (Kaushty) [reply README.md]: Addressed.

File: README.md (hunk 5 above)
Comment (Kaushty) [reply README.md]: Addressed.

**Review by rijil-tr — APPROVED** [review]
LGTM

**Review by naveenctstack — APPROVED** [review]

**Review by aravindbuilt — APPROVED** [review]

**Pull request merged**